src/main/java/com/feedzai/commons/sql/abstraction/engine/impl/OracleEngine.java

```diff
@@ -160,6 +164,38 @@ protected int entityToPreparedStatement(final DbEntity entity, final PreparedSta
         return i - 1;
     }
 
+    @Override
+    protected void setParameterValues(final PreparedStatement ps, final int index, final Object param) throws SQLException {
+        if (param instanceof byte[]) {
+            ps.setBytes(index, (byte[]) param);
+        } else {
+            if (!(param instanceof Double)) {
+                /**
+                 * If it is not a double it may be:
+                 *  - A String with the special values such as infinity or NaN
+                 *  - A String with some invalid value
+                 *  - An integer/long
+                 *  - Some other invalid type
+                 *
+                 *  In these cases we will invoke the setObject.
+                 */
+                ps.setObject(index, param);
+            } else {
+                /**
+                 * Only Double typed values will execute here.
+                 * They may be:
+                 *  - Regular double number
+                 *  - Double.NaN
+                 *  - Double.{POSITIVE|NEGATIVE}_INFINITY
+                 *
+                 *  In these cases we use the setBinaryDouble specific to the oracle prepared statement.
+                 */
+                final OraclePreparedStatement orclPs = (OraclePreparedStatement) ps;
+                orclPs.setBinaryDouble(index, (Double) ensureNoUnderflow(param));
```

Comment:
Does this work in the old native type for double ?  If not, maybe we can use setBinaryDouble() only if the value is NaN or Infinity, and use setObject().